host/integrationbase.go

```diff
@@ -182,7 +180,7 @@ func (s *IntegrationBase) setupSuiteForPinotTest() {
 }
 
 func (s *IntegrationBase) setupLogger() {
-	s.Logger = testlogger.New(s.T())
+	s.Logger = loggerimpl.NewLoggerForTest(s.Suite)
```

Comment:
sadly another:
```
panic: Log in goroutine after TestIntegrationSuite has completed: 2023-12-19T00:11:34.305Z	WARN	Async task dispatch timed out	{"component": "matching-engine", "wf-domain-name": "integration-test-domain-66697d18-7114-4288-806e-80331e8c9c86", "wf-task-list-name": "integration-signal-with-start-workflow-id-reuse-test-tasklist", "wf-task-list-type": 0, "isolation-group": "", "wf-run-id": "c29381b5-fdf8-4357-b9da-a022f109dee7", "wf-id": "integration-signal-with-start-workflow-id-reuse-test", "queue-task-id": 1, "wf-domain-id": "6a5ef3dd-228b-4424-96f3-34517d523af9", "logging-call-at": "taskReader.go:418"}
goroutine 90361 [running]:
testing.(*common).logDepth(0xc000fe7ba0, {0xc000c886c0, 0x21d}, 0x3)
	/usr/local/go/src/testing/testing.go:1003 +0x68c
testing.(*common).log(...)
	/usr/local/go/src/testing/testing.go:985
testing.(*common).Logf(0xc000fe7ba0, {0x6202a90, 0x2}, {0xc0017e8260, 0x1, 0x1})
	/usr/local/go/src/testing/testing.go:1036 +0xa5
go.uber.org/zap/zaptest.testingWriter.Write({{0x62411f0?, 0xc000fe7ba0?}, 0xe0?}, {0xc0046f5c00, 0x21e, 0x400})
	/go/pkg/mod/go.uber.org/zap@v1.13.0/zaptest/logger.go:130 +0x12d
go.uber.org/zap/zapcore.(*ioCore).Write(0xc002835da0, {0x1, {0xc15855cd92375e0e, 0x81de9b65e1, 0x7ddfec0}, {0x0, 0x0}, {0x4b0ecef, 0x1d}, {0x0, ...}, ...}, ...)
	/go/pkg/mod/go.uber.org/zap@v1.13.0/zapcore/core.go:90 +0x19a
go.uber.org/zap/zapcore.(*CheckedEntry).Write(0xc00180ca50, {0xc0003f5180, 0x6, 0xa})
	/go/pkg/mod/go.uber.org/zap@v1.13.0/zapcore/entry.go:216 +0x2ed
go.uber.org/zap.(*Logger).Warn(0xc003d4b3e0?, {0x4b0ecef, 0x1d}, {0xc0003f5180, 0x6, 0xa})
	/go/pkg/mod/go.uber.org/zap@v1.13.0/logger.go:195 +0x6e
github.com/uber/cadence/common/log/loggerimpl.(*loggerImpl).Warn(0xc003d4b3e0, {0x4b0ecef, 0x1d}, {0xc00016f040, 0x5, 0x5})
	/cadence/common/log/loggerimpl/logger.go:138 +0x18f
github.com/uber/cadence/service/matching.(*taskReader).dispatchSingleTaskFromBuffer(0xc002a8bcc0, {0x0, 0x0}, 0xc0052fa900)
	/cadence/service/matching/taskReader.go:418 +0x123d
github.com/uber/cadence/service/matching.(*taskReader).dispatchSingleTaskFromBufferWithRetries(...)
	/cadence/service/matching/taskReader.go:389
github.com/uber/cadence/service/matching.(*taskReader).dispatchBufferedTasks(0xc002a8bcc0, {0x0, 0x0})
	/cadence/service/matching/taskReader.go:159 +0x2c5
created by github.com/uber/cadence/service/matching.(*taskReader).Start
	/cadence/service/matching/taskReader.go:126 +0x171
FAIL	github.com/uber/cadence/host	557.826s
```
since these are so similar I just reverted all of `host`.

Reply:
taskReader's `Stop()` doesn't wait for these goroutines to complete. We can add a waitgroup to avoid this leak

Reply:
That will generate around 40Mb of logs, so I would prefer introducing a fix instead of hacking around. Let's land the revert and I'll work on the fix.

Reply:
Addressed this leak in https://github.com/uber/cadence/pull/5499

Reply:
Should be resolved. Please revert and let me know if there are more leaking parts.

Reply:
undid (just copied master) and tests pass on the first try - thanks!